lib/rouge/lexers/jinja.rb

```diff
@@ -133,6 +138,15 @@ def self.word_operators
 
         rule /\%\}/, Comment::Preproc, :pop!
       end
+
+      state :raw do
+        rule %r{(\{\%)(\s+)(endverbatim|endraw)(\s+)(\%\})} do
```

Comment:
you would need separate states for verbatim and raw actually. Otherwise, opening with `{% verbatim %}` could be closed with `{% endraw %}` which is not the case.

Reply:
That's true for this to be fully correct. But for the sake of simplicity, we left open that possibility. It's a non-goal of Rouge to check the correctness of input. It's enough that syntactically correct code is highlighted correctly :)

(I think I said the word 'correct' too many times in there :P)

Reply:
the thing is, anything inside a `{% verbatim %}` tag is syntactically correct, including `{% endraw %}`.

Reply:
Hmmm. So you mean a situation where someone would use `{% verbatim %}` to escape `{% raw %}`?

Reply:
yes, that could happen

Reply:
Fair enough. If you can open an issue, that'd be helpful for future revisions :)